cli/command/image/save.go

```diff
@@ -54,3 +61,14 @@ func runSave(dockerCli command.Cli, opts saveOptions) error {
 
 	return command.CopyToFile(opts.output, responseBody)
 }
+
+func validateOutputPath(path string) error {
+	dir := filepath.Dir(path)
+	if dir != "" && dir != "." {
+		_, err := os.Stat(dir)
+		if err != nil && os.IsNotExist(err) {
+			return errors.New("directory does not exist")
```

Comment:
A custom error was used because the returned error from `os.Stat` is:
```
GetFileAttributesEx fakedir: The system cannot find the file specified.
```
which is less useful imo (other than showing the stat'd file name).

```
fmt.Errorf("directory %q does not exist", dir)
```
should fix the issue that the dir name is not shown while still being a better error message. WDYT?